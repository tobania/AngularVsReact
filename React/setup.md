# React Setup

## Prerequirements
- nodejs
- Backend runs in the background and is accessable

# Setup
Navigate to the frontend root (src) and execute the following commands:
npm install
npm start
