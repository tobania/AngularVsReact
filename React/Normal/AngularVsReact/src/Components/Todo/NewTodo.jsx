import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import CreateTodo from '../../HttpServices/Todos/CreateTodoService';

class NewTodo extends PureComponent {
  constructor() {
    super();
    this.state = { title: '' };

    this.handleChange = this.handleChange.bind(this);
    this.addTodo = this.addTodo.bind(this);
  }

  handleChange(event) {
    this.setState({ title: event.target.value });
  }

  addTodo() {
    const { title } = this.state;
    const { addCreatedTodo } = this.props;
    CreateTodo({ title }, addCreatedTodo);
  }

  render() {
    const { title } = this.state;

    return (
      <div className="col-md-3">
        <div className="card shadow mb-4">
          <div className="card-header">
            <div className="row">
              <div className="col-9">
                <input type="text" className="form-control" value={title} onChange={this.handleChange} />
              </div>
              <div className="offset-1 col-2">
                <button type="button" className="btn btn-success float-right" onClick={this.addTodo}>Save</button>
              </div>
            </div>
          </div>
          <div className="card-body" />
        </div>
      </div>
    );
  }
}

NewTodo.propTypes = {
  addCreatedTodo: PropTypes.func.isRequired,
};

NewTodo.defaultProps = {
};

export default NewTodo;
