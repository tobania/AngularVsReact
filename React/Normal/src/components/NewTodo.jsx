import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NewTodo extends Component {
  constructor() {
    super();
    this.state = { title: '' };

    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.setState({ title: event.target.value });
  }

  render() {
    const { addTodo } = this.props;
    const { title } = this.state;

    return (
      <div className="col-md-3">
        <div className="card shadow mb-4">
          <div className="card-header">
            <div className="row">
              <div className="col-9">
                <input type="text" className="form-control" value={title} onChange={this.handleChange} />
              </div>
              <div className="offset-1 col-2">
                <button type="button" className="btn btn-success float-right" onClick={() => addTodo(title)}>Save</button>
              </div>
            </div>
          </div>
          <div className="card-body" />
        </div>
      </div>
    );
  }
}

NewTodo.propTypes = {
  addTodo: PropTypes.func.isRequired,
};

NewTodo.defaultProps = {
};

export default NewTodo;
