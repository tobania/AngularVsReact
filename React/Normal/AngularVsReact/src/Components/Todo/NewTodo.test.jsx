import React from 'react';
import renderer from 'react-test-renderer';
import NewTodo from './NewTodo';

describe('NewTodo', () => {
  it('gets loaded correctly', () => {
    const component = renderer.create(
      <NewTodo addCreatedTodo={jest.fn} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handle change should update the state', () => {
    const component = renderer.create(
      <NewTodo addCreatedTodo={jest.fn} />,
    ).getInstance();

    expect(component.state.title).toBe('');
    component.handleChange({ target: { value: 'test' } });
    expect(component.state.title).toBe('test');
  });

  it('addTodo can be called', () => {
    const component = renderer.create(
      <NewTodo addCreatedTodo={jest.fn} />,
    ).getInstance();

    component.addTodo();
  });
});
