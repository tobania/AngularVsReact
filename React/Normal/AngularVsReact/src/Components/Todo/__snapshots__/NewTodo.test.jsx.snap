// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodo gets loaded correctly 1`] = `
<div
  className="col-md-3"
>
  <div
    className="card shadow mb-4"
  >
    <div
      className="card-header"
    >
      <div
        className="row"
      >
        <div
          className="col-9"
        >
          <input
            className="form-control"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="offset-1 col-2"
        >
          <button
            className="btn btn-success float-right"
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
        </div>
      </div>
    </div>
    <div
      className="card-body"
    />
  </div>
</div>
`;
